src/app/routes/article/article.service.ts

```diff
@@ -379,7 +410,15 @@ export const updateArticle = async (article: any, slug: string, id: number) => {
     },
   });
 
-  return articleMapper(updatedArticle, id);
+  // Invalidate old slug cache
```

Comment:
Since the cache is pretty simple and the cache key is basically the article slug it would lend it self to a more cleaner implementation in a form of a wrapper which wraps around the service (implements the same interface) or an http middleware. In both cases the cache would not leak into the service but would instead intercept the request and set and invalidate cache as needed.